src/main/java/com/epam/eco/commons/kafka/consumer/bootstrap/BootstrapConsumerExecutor.java

```diff
@@ -62,6 +63,17 @@ public void waitForBootstrap() throws InterruptedException {
         }
     }
 
+    public boolean waitForBootstrap(long timeout, TimeUnit timeUnit) throws InterruptedException {
+        boolean result = true;
+        for (BootstrapConsumerThread<K, V, R> thread : threads) {
```

Comment:
Let's optimize the code to avoid useless waiting
`    for (BootstrapConsumerThread<K, V, R> thread : threads) {
            if (!thread.waitForBootstrap(timeout, timeUnit)) {
                return false;
            }
        }
    return true;`